test/check/check.go

```diff
@@ -353,6 +353,26 @@ func checkForDeployment(name string) error {
 	return nil
 }
 
+// CheckOperatorPodStability makes sure that the CNAO pod has not restarted since it started working
+func CheckOperatorPodStability() {
+	pods := corev1.PodList{}
+	listOptions := client.ListOptions{}
+	labelSelector, err := k8slabels.Parse(fmt.Sprintf("name, name == %s", components.Name))
+	Expect(err).NotTo(HaveOccurred(), "should succeed parsing label selector")
+	listOptions.LabelSelector = labelSelector
```

Comment:
Selecting labels with controller-runtime is easier
```golang
import "sigs.k8s.io/controller-runtime/pkg/client"
List(ontext.Background(), &pods, &client.MatchingLabels(map[string]string{})) 		
```

Reply:
I think `frameworkClient` List needs ListOptions, while : 
```
func (f *frameworkClient) List(gCtx goctx.Context, list runtime.Object, opts ...dynclient.ListOption)
```

Reply:
ListOptions is an interface and MatchingLabels implement it https://github.com/kubernetes-sigs/controller-runtime/blob/e63948755ba14b3dd8506fafbef4065fb9457817/pkg/client/options.go#L41